TOOLS/ce/src/main/scala/org.opalj.ce/ConfigObject.scala

```diff
@@ -0,0 +1,171 @@
+     * @param HTMLHeadline accepts the HTML syntax of the Headline of the value. Can contain $ label and $ brief flags for filling with content.
+     * @param HTMLContent accepts the HTML syntax of the content frame for the value. Must contains a $ content flag for correct rendering.
+     * @return returns the Config Object as HTML code.
+     */
+    override def toHTML(label: String, HTMLHeadline: String, HTMLContent: String, sorted: Boolean): String = {
+        var HTMLString = ""
+        var head = label
+        if (this.comment.label.nonEmpty) head = this.comment.label
+
+        // Get HTML data for all child Nodes
+        var content = "<p>" + comment.toHTML + "</p>"
+
+        if (sorted) {
+            val sortedKeys = entries.keys.toSeq.sorted
+
+            for (key <- sortedKeys) {
+                content += entries(key).toHTML(key, HTMLHeadline, HTMLContent, sorted)
+            }
+        } else {
+            for ((key, node) <- entries) {
+                content += node.toHTML(key, HTMLHeadline, HTMLContent, sorted)
+            }
+        }
+
+        // Adds Header line with collapse + expand options
+        HTMLString += HTMLHeadline.replace("$label", head).replace("$brief", this.comment.getBrief)
+
+        // Add content below
+        HTMLString += HTMLContent.replace("$content", content)
+
+        HTMLString
+    }
+
+    /**
+     * Checks if the object is empty
+     * @return true if both the Object and the comment are empty
+     */
+    override def isEmpty: Boolean = {
+        if (!comment.isEmpty) return false
+        for ((key, value) <- entries) {
+            if (!value.isEmpty) return false
+        }
+        true
+    }
+
+    /**
+     * Merges two type compatible objects.
+     * This means that the objects are free of conflicting values and lists. Objects are allowed to overlap as long as there are no conflicts down the tree.
+     * @param insertingObject Is the object that is supposed to be merged into the executing one
+     */
+    def merge(insertingObject: ConfigObject): Unit = {
+
+        // Expanding both objects guarantees compatible key naming syntax
+        this.expand()
+        insertingObject.expand()
+
+        // Insert object
+        for (kvpair <- insertingObject.entries) {
+            val (key, value) = kvpair
+            if (this.entries.contains(key)) {
+                val conflicting_entry = this.entries.getOrElse(key, null)
+                if (conflicting_entry.isInstanceOf[ConfigObject] && value.isInstanceOf[ConfigObject]) {
+                    val conflicting_child_object = conflicting_entry.asInstanceOf[ConfigObject]
+                    conflicting_child_object.merge(value.asInstanceOf[ConfigObject])
+                } else {
+                    println("Info on incompatible keys: " + key.trim)
+                    throw new Exception(
```

Comment:
Don't thow plain exceptions, use a reasonable exception type (if even necessary at all and can't just handled with a log entry).

Reply:
Changed Exception type to IllegalArgument exception.